roles/postgres/tasks/setup_db.yml

```diff
@@ -0,0 +1,63 @@
+---
+- name: Include OS specific variables
+  ansible.builtin.include_vars: "{{ item }}"
+  loop: >-
+    {{ lookup('first_found', os_fallback, errors='ignore', wantlist=True) }}
+
+- name: Configure postgresql client auth
+  become: true
+  notify:
+    - Restart-postgresql
+  ansible.builtin.blockinfile:
+    path: "{{ postgresql_conf_path }}/pg_hba.conf"
+    block: |
+      {% for host in postgres_db_clients | map('extract', hostvars, ['ansible_default_ipv4', 'address']) %}
+      host {{ postgres_db_name }} {{ postgres_db_username }} {{ host }}/32 md5
+      {% endfor %}
+    marker: >-
+      # {mark} ANSIBLE MANAGED: allow {{ postgres_db_clients | join(", ") }} to connect to {{ postgres_db_name }} as {{ postgres_db_username }}
```

Comment:
is it ok that the marker contains variables? isn't gonna mess updates if those variables changes on a next run?

Reply:
I'm not sure what behaviour we'd really want in such case. With current code, changing the clients, username or db name would result in additional entries while with no variable in would replace the old one with the new one but would also prevent passing multiple databases unless we keep at least the db name in the marker. Then still, it would not be possible to pass multiple users/clients entries for the same db....
All in all I think the current approach is better than other that bring these limitations.

Reply:
not following you, I was looking that you used variables inside a marker, but looking at the docs looks like that `{mark}` is the only thing really used by ansible to recognize the block

Reply:
`{mark}` evaluate either to `BEGIN` or `END` depending on whether it marks the beginning or the end of the block.

So if you call the `setup_db` endpoint twice with 2 different dbs or/and hosts or/and users and/or clients, but the mark is the same for both iterations, the second block will replace the first one